### PR TITLE
Work correctly if iOS runtime version is lower than 1.3

### PR DESCRIPTION
**Pull request by Fatme**
Fixes https://github.com/NativeScript/nativescript-cli/issues/832

### CODE REVIEW

File: lib/services/android-project-service.ts [hunk 1]
```diff
@@ -64,6 +63,12 @@ class AndroidProjectService extends projectServiceBaseLib.PlatformProjectService
 		return this._platformData;
 	}
 
+	public getAppResourcesDestinationDirectoryPath(): IFuture<string> {
+		return (() => {
```
Comment (teobugslayer) [inline lib/services/android-project-service.ts]: You may use future.fromResult\<string\> here

File: lib/services/ios-project-service.ts [hunk 2]
```diff
@@ -5,6 +5,7 @@ import * as path from "path";
 import * as shell from "shelljs";
 import * as util from "util";
 import * as os from "os";
+import * as semver from "semver";
```
Comment (teobugslayer) [inline lib/services/ios-project-service.ts]: Are you sure this work? semver is exported as 
`
declare module "semver" {
	export = SemVerModule;
}
`
and I expect that only imports with require work.

File: lib/services/ios-project-service.ts [hunk 3]
```diff
@@ -61,6 +62,19 @@ export class IOSProjectService extends projectServiceBaseLib.PlatformProjectServ
...
+	public getAppResourcesDestinationDirectoryPath(): IFuture<string> {
+		return (() => {
+			this.$projectDataService.initialize(this.$projectData.projectDir);
+			let frameworkVersion = this.$projectDataService.getValue(this.platformData.frameworkPackageName).wait()["version"];
+
+			if(semver.lt(frameworkVersion, "1.3.0")) {
+				return path.join(this.platformData.projectRoot, this.$projectData.projectName, "Resources", "icons");
+			}
+
+			return path.join(this.platformData.projectRoot, this.$projectData.projectName, "Resources");
```
Comment (teobugslayer) [inline lib/services/ios-project-service.ts]: Both return statements build the same path - is this expected?

File: lib/services/ios-project-service.ts [hunk 4]
```diff
@@ -265,7 +279,9 @@ export class IOSProjectService extends projectServiceBaseLib.PlatformProjectServ
 	}
 
 	public prepareAppResources(appResourcesDirectoryPath: string): IFuture<void> {
-		return this.$fs.deleteDirectory(this.platformData.appResourcesDestinationDirectoryPath);
+		return (() => {
```
Comment (teobugslayer) [inline lib/services/ios-project-service.ts]: You can unwrap the return -> wait -> future chain

File: lib/services/ios-project-service.ts (hunk 3 above)
Comment (Fatme) [inline lib/services/ios-project-service.ts]: The paths are different - `path.join(this.platformData.projectRoot, this.$projectData.projectName, "Resources", "icons");` and `path.join(this.platformData.projectRoot, this.$projectData.projectName, "Resources");`

File: lib/services/ios-project-service.ts (hunk 4 above)
Comment (Fatme) [inline lib/services/ios-project-service.ts]: I have `.wait` inside the function

File: lib/services/ios-project-service.ts (hunk 3 above)
Comment (teobugslayer) [inline lib/services/ios-project-service.ts]: Oh dear, the last part ("icons") went on another line on my monitor and I missed it completely. I am so sorry.

File: lib/services/ios-project-service.ts (hunk 4 above)
Comment (teobugslayer) [inline lib/services/ios-project-service.ts]: And I feel stupid :(

**Comment by teobugslayer** [comment]
tests are failing:
```
Npm support tests
TypeError: Object #<Object> has no method 'getAppResourcesDestinationDirectoryPath'
    at FiberFuture.Future.wait (/home/travis/build/NativeScript/nativescript-cli/node_modules/fibers/future.js:535:15)
    at Context.<anonymous> (/home/travis/build/NativeScript/nativescript-cli/test/npm-support.js:145:39)
    at /home/travis/build/NativeScript/nativescript-cli/node_modules/mocha-fibers/lib/mocha-fibers.js:25:14
    - - - - -
    at /home/travis/build/NativeScript/nativescript-cli/lib/services/platform-service.js:133:81
    at Function.settle (/home/travis/build/NativeScript/nativescript-cli/node_modules/fibers/future.js:249:26)
    at FiberFuture.Future.wait (/home/travis/build/NativeScript/nativescript-cli/node_modules/fibers/future.js:534:10)
    at Context.<anonymous> (/home/travis/build/NativeScript/nativescript-cli/test/npm-support.js:145:39)
    at /home/travis/build/NativeScript/nativescript-cli/node_modules/mocha-fibers/lib/mocha-fibers.js:25:14
```

**Comment by teobugslayer** [comment]
:+1:

**Pull request merged**